app/models/sample.rb

```diff
@@ -472,6 +472,10 @@ def sample_empty?(supplier_sample_name = name)
     sample_supplier_name_empty?(supplier_sample_name)
   end
 
+  def supplier
+    sample_manifest ? sample_manifest.supplier.name : nil
```

Comment:
Use the safe navigation operator:
```
sample_manifest&.supplier&.name
```

Reply:
Or we could actually add:
```
has_one :supplier, through: :sample_manifest
```
To pop a direct link to the supplier object. Normally you'd then expose the name as `supplier_name` although that would cause confusion here due to the confusingly named attribute in sample metadata.

I'm not sure quite what to suggest in terms of naming.